### PR TITLE
PR to use 3-digit evs_ver, use $MAILTO, remove pgmout, etc ...

### PR DESCRIPTION
**Pull request by ShelleyMelchior-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>
This PR updates evs_ver to be 3-digits. Impacted files are `verisons/run.ver` and all files in `dev/drivers/scripts/`
This PR removes the `sorc/` script that built the symlink to the `fix/` directory.
This PR removes any lingering instances of pgmout.
This PR replaces $maillist w/ the NCO Table 1 variable, $MAILTO.
This PR moves `modulefiles/` to `dev/modulefiles/`. Impacted files are all files in `dev/drivers/scripts/`

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>
Testing should include random selection of components for all 3 steps to assure EVS-wide impact is not harmful.
I also suggest one end-to-end test for all 3 (or 2) steps in a single component.

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
Standby on testing this PR until I can move the `modulefiles/` to `dev/`.

**Comment by ShelleyMelchior-NOAA** [comment]
Further standby until I can update the driver scripts to use the new location for the modulefiles that get sourced.

**Comment by ShelleyMelchior-NOAA** [comment]
🚥 green light to proceed!

**Comment by ShelleyMelchior-NOAA** [comment]
@PerryShafran-NOAA , I suggest analyses plots to be one of your tests.

**Comment by PerryShafran-NOAA** [comment]
Just FYI - I am starting with the naefs prep script in global_ens.  I made the change to the gefs_mbrs statement by removing it from the driver and putting it into the exevs script.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA I completed the naefs prep step.  It looks like the modulefile was correctly referenced.  

I do notice that you have the 3-digit evs_ver in the run.ver file.  You also have it in the main driver script, still:
 
export evs_ver=v1.0.0
evs_ver_2d=$(echo $evs_ver | cut -d'.' -f1-2)

Are we keeping the evs_ver in the driver script for now, since it's in dev and we don't care?  

Anyway, the output directory is also good:  /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/global_ens/atmos.20231115 

(Only the cmce_bc and gefs_bc directories should be populated with the naefs prep job, the other directories would be populated with the global_ens prep step, which takes much longer to test, so I'm going to dispense with that test.  

I'm not sure how to test the mailto functionality.  I guess since it's just a variable that it doesn't matter whether it's $MAILTO or $maillist, it should work exactly the same?  

Anyway, I'm happy with the prep step, if you address the question about evs_ver.

File: dev/drivers/scripts/prep/global_ens/jevs_global_ens_naefs_atmos_prep.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ export MODELNAME=naefs
 source $HOMEevs/versions/run.ver
 module reset
 module load prod_envir/${prod_envir_ver}
-source $HOMEevs/modulefiles/${COMPONENT}/${COMPONENT}_${STEP}.sh
+source $HOMEevs/dev/modulefiles/${COMPONENT}/${COMPONENT}_${STEP}.sh
 
 export evs_ver=v1.0.0
```
Comment (AliciaBentley-NOAA) [inline dev/drivers/scripts/prep/global_ens/jevs_global_ens_naefs_atmos_prep.sh]: @ShelleyMelchior-NOAA Please remove export evs_ver=v1.0.0 in [dev/drivers/scripts/prep/global_ens/jevs_global_ens_naefs_atmos_prep.sh](https://github.com/NOAA-EMC/EVS/pull/354/files#diff-56df572888001debd41ed0a7e26b47fa52512b0c4ddbc2155d0f66525d0e143d). 

CC @PerryShafran-NOAA

**Comment by PerryShafran-NOAA** [comment]
Running jevs_rtofs_osisaf_stats.sh for a stats test.

**Comment by ShelleyMelchior-NOAA** [comment]
Ok. I see what happened. The `export evs_ver=v1.0.0` got reintroduced in _**some**_ components when I merged in develop after PRs were merged. I also see that build.ver still uses the 2-digit version for evs_ver. Let me take care of that clean up.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Will I need to re-run what's been done above?

**Comment by ShelleyMelchior-NOAA** [comment]
There were many components where `export evs_ver` snuck back into the drivers. I would retest prep, maybe choose a different component this time.

I also just made changes to make sure no more $maillist vars snuck back in.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA @ShelleyMelchior-NOAA Since build.ver was changed, can we also compile and test something with this PR?

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA excellent idea!  I'll do the global_det prep, which I believe uses a lot of these codes.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA The global_det prep step completed, and looked like the executables are all OK.  See /lfs/h2/emc/vpppg/noscrub/perry.shafran/pr354test/EVS/dev/drivers/scripts/prep/global_det/jevs_global_det_atmos_prep_00.o93488665.  But you will see WARNINGs due to missing files, but that's also because I'm running too early to catch some stuff that comes in later.  I'm doing a re-run with PDYm2.  

Also see the output directory /lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/prep/global_det/atmos.20231116

Feel that in general this step worked pretty well as intended.

**Comment by ShelleyMelchior-NOAA** [comment]
Thanks Perry. global_det prep seems OK. ✔️

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Good news!  I made Marcel's suggested change to the analysis plot job, and it changed the word ERROR to WARNING.

If you want to add this change, please grab my file and add to this branch:

/lfs/h2/emc/vpppg/noscrub/perry.shafran/pr354test/EVS/ush/analyses/df_preprocessing.py

And also you may validate:

The .o file is here:

/lfs/h2/emc/vpppg/noscrub/perry.shafran/pr354test/EVS/dev/drivers/scripts/plots/analyses/jevs_analyses_grid2obs_plots.o93490085

And the plot tarball is here:

/lfs/h2/emc/vpppg/noscrub/perry.shafran/evs/v1.0/plots/analyses/atmos.20231115

Not sure why this wound up in noscrub?  The plots are going to ptmp in the emc.vpppg correctly.  Anyway, here you go....

**Comment by PerryShafran-NOAA** [comment]
I made the updates directly into your code.

**Comment by PerryShafran-NOAA** [comment]
We have determined that emails are being sent using the mailto feature, by running a job and choosing a date that doesn't have any current data in production.  

I also did an nfcens test, since that's what was just merged in.  That looks good too.

We may be ready to merge!

File: dev/drivers/scripts/plots/global_ens/jevs_global_ens_headline_gefs_grid2grid_plots.sh [hunk 2]
```diff
@@ -21,13 +21,12 @@ export COMPONENT=global_ens
 export VERIF_CASE=grid2grid
 export MODELNAME=gefs
 
-source $HOMEevs/modulefiles/${evs_ver_2d}
+source $HOMEevs/dev/modulefiles/${evs_ver_2d}
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/plots/global_ens/jevs_global_ens_headline_gefs_grid2grid_plots.sh]: @PerryShafran-NOAA  ... what's going on here? I think this line needs to go.

File: ush/global_ens/evs_get_gens_atmos_data.sh [hunk 3]
```diff
@@ -657,20 +657,26 @@ if [ $modnam = osi_saf ] ; then
...
+    fi 
+=======
+          cat mailmsg | mail -s "$subject" $MAILTO
          else
              python ${USHevs}/global_ens/global_ens_sea_ice_prep.py
              [[ $SENDCOM="YES" ]] &&  cpreq -v $WORK/atmos.${INITDATE}/osi_saf/*.nc $COMOUTosi_saf/.
          fi
    fi
+>>>>>>> develop
 fi
```
Comment (PerryShafran-NOAA) [inline ush/global_ens/evs_get_gens_atmos_data.sh]: @ShelleyMelchior-NOAA - what happened here?  Can we clean this up ?

File: dev/drivers/scripts/plots/global_ens/jevs_global_ens_headline_gefs_grid2grid_plots.sh (hunk 2 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/plots/global_ens/jevs_global_ens_headline_gefs_grid2grid_plots.sh]: @ShelleyMelchior-NOAA You mean the existence of the global_ens plot job?  

Yeah the line is funky, but this file isn't going into para.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/plots/global_ens/jevs_global_ens_headline_gefs_grid2grid_plots.sh]: @ShelleyMelchior-NOAA Can you fix?

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Thanks for making those fixes!  I think we are ok for merge, yes?

@AliciaBentley-NOAA

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA I think this is ready to merge, yes! @ShelleyMelchior-NOAA can confirm.

**Comment by ShelleyMelchior-NOAA** [comment]
I approve this PR for merge. ✔️

**Review by PerryShafran-NOAA — APPROVED** [review]
Approved for merge.

**Pull request merged**